lib/Majkl578/NetteAddons/Doctrine2Identity/DI/IdentityExtension.php

```diff
@@ -25,11 +25,11 @@ class IdentityExtension extends CompilerExtension
 {
 	const NAME = 'doctrine2identity';
 
-	public function loadConfiguration()
+	public function beforeCompile()
 	{
 		$builder = $this->getContainerBuilder();
-
-		$builder->getDefinition('nette.userStorage')
+		$builder->removeDefinition($builder->getByType('Nette\Security\IUserStorage') ?: 'nette.userStorage');
```

Comment:
@Northys Because you need to set factory, not class. Check code of `SecurityExtension` - https://github.com/nette/security/blob/master/src/Bridges/SecurityDI/SecurityExtension.php#L44

Calling following works in Nette 2.3 and you don't have to remove definition
```php
$builder->getDefinition('security.userStorage')
	->setFactory('Majkl578\NetteAddons\Doctrine2Identity\Http\UserStorage');
```